src/exec_auth.ts

```diff
@@ -4,12 +4,25 @@ import { Authenticator } from './auth';
 import { User } from './config_types';
 
 export class ExecAuth implements Authenticator {
+    private tokenCache: any = {};
+
     public isAuthProvider(user: User) {
         return user.authProvider.name === 'exec' ||
             (user.authProvider.config && user.authProvider.config.exec);
     }
 
     public getToken(user: User): string | null {
+        // TODO: Handle client cert auth here, requires auth refactor.
+        // See https://kubernetes.io/docs/reference/access-authn-authz/authentication/#input-and-output-formats
+        // for details on this protocol.
+        const cachedToken = this.tokenCache[user.name];
+        if (cachedToken) {
+            const date = Date.parse(cachedToken.status.expirationTimestamp);
+            if (date < Date.now()) {
+                return cachedToken.status.token;
+            }
+            this.tokenCache[user.name] = null;
```

Comment:
Maybe I am missing some context here but I can't see where `tokenCache` is getting populated?

Reply:
Fixed.

Reply:
Fixed. (good catch)